src/core/data.service.ts

```diff
@@ -43,18 +44,18 @@ export class DataService {
   /**
    * Notifies any data change listeners about a data change.
    *
-   * @param {ControlElement} uischema the affected control element
+   * @param {ControlElement} controlElement the affected control element
    * @param newValue the new value of the data chunk
    */
-  notifyChange(uischema: ControlElement, newValue: any): void {
-    if (uischema !== undefined && uischema !== null) {
-      const pair = getValuePropertyPair(this.data, uischema.scope.$ref);
+  notifyDataChangeListeners(controlElement: ControlElement, newValue: any): void {
```

Comment:
In my opinion this name is not correct, as we are calling this on the dataservice in order to notify the dataservice about a datachange, so if a control renderer changes a value it notifies the dataservice.
So a better name would be notifyAboutDataChange ?